server/dive_tasks/tasks.py

```diff
@@ -297,40 +279,36 @@ def train_pipeline(
         return
 
     gc: GirderClient = self.girder_client
+    utils.authenticate_urllib(gc)
     manager.updateStatus(JobStatus.FETCHING_INPUT)
 
+    # Extract params
+    results_folder_id = params['results_folder_id']
+    dataset_input_list = params['dataset_input_list']
+    pipeline_name = params['pipeline_name']
+    config = params['config']
+    annotated_frames_only = params['annotated_frames_only']
+
     pipeline_base_path = Path(conf.get_extracted_pipeline_path())
     config_file = pipeline_base_path / config
-
-    if len(source_folder_list) != len(groundtruth_list):
-        raise Exception("Ground truth doesn't exist for all folders")
-
-    # List of folderIds used for training
-    trained_on_list: List[str] = []
-    # List of[input folder / ground truth file] pairs for creating input lists
+    # List of (input folder, ground truth file) pairs for creating input lists
     input_groundtruth_list: List[Tuple[Path, Path]] = []
     # root_data_dir is the directory passed to `viame_train_detector`
     with tempfile.TemporaryDirectory() as _working_directory, suppress(utils.CanceledError):
         _working_directory_path = Path(_working_directory)
         input_path = utils.make_directory(_working_directory_path / 'input')
         output_path = utils.make_directory(_working_directory_path / 'output')
 
-        for index in range(len(source_folder_list)):
-            source_folder = source_folder_list[index]
-            groundtruth = groundtruth_list[index]
-            download_path = input_path / source_folder['name']
-            trained_on_list.append(str(source_folder["_id"]))
+        for source_folder_id, revision in dataset_input_list:
+            download_path = input_path / source_folder_id
```

Comment:
You need to actually make this directory, probably with `utils.make_directory(input_path / source_folder_id)`

Reply:
I don't need to, but it might make the code more robust.  This path is passed to `download_source_media`, which will create the path when the download happens.

Reply:
It's fixed by creating the folder, but it wasn't about the `download_source_media`,  that did properly create the folder.   It was about the `utils.download_revision_csv` which is based on `groundtruth_path = download_path / 'groundtruth.csv'` and download_path as a folder doesn't exist yet.  The urlretrieve fails to place the groundtruth into the non-existent folder.  You could instead change `download_revision_csv` to auto-ensure the path exists by creating it if necessary, but I think explicit creation is probably better.

Reply:
Ah, OK.  I added it anyway.  Now CI is failing mysteriously.